In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import re
import numpy as np


In [ ]:
path = '/content/gdrive/My Drive/2025_dacon_etri/data/'

train_file_name1 = "submission_0.6350958281670923_lightgbm.csv"
submission_df = pd.read_csv(path+train_file_name1, encoding='utf8')

In [ ]:
submission_df.rename(columns = {'Q1' : 'Target1_Q1'}, inplace = True)
submission_df.rename(columns = {'Q2' : 'Target1_Q2'}, inplace = True)
submission_df.rename(columns = {'Q3' : 'Target1_Q3'}, inplace = True)
submission_df.rename(columns = {'S1' : 'Target1_S1'}, inplace = True)
submission_df.rename(columns = {'S2' : 'Target1_S2'}, inplace = True)
submission_df.rename(columns = {'S3' : 'Target1_S3'}, inplace = True)

In [ ]:
submission_df.head()

,subject_id,sleep_date,lifelog_date,Target1_Q1,Target1_Q2,Target1_Q3,Target1_S1,Target1_S2,Target1_S3
0,id01,2024-07-31,2024-07-30,0,1,0,1,1,1
1,id01,2024-08-01,2024-07-31,0,1,0,0,1,1
2,id01,2024-08-02,2024-08-01,0,0,1,0,0,1
3,id01,2024-08-03,2024-08-02,1,0,0,1,1,1
4,id01,2024-08-04,2024-08-03,1,1,1,1,1,1


In [ ]:
train_file_name2 = "submission_0.6359383853269983_xgboost.csv"
submission_df2 = pd.read_csv(path+train_file_name2, encoding='utf8')

In [ ]:
submission_df2.rename(columns = {'Q1' : 'Target2_Q1'}, inplace = True)
submission_df2.rename(columns = {'Q2' : 'Target2_Q2'}, inplace = True)
submission_df2.rename(columns = {'Q3' : 'Target2_Q3'}, inplace = True)
submission_df2.rename(columns = {'S1' : 'Target2_S1'}, inplace = True)
submission_df2.rename(columns = {'S2' : 'Target2_S2'}, inplace = True)
submission_df2.rename(columns = {'S3' : 'Target2_S3'}, inplace = True)

In [ ]:
submission_df2.head()

,subject_id,sleep_date,lifelog_date,Target2_Q1,Target2_Q2,Target2_Q3,Target2_S1,Target2_S2,Target2_S3
0,id01,2024-07-31,2024-07-30,0,1,0,1,1,1
1,id01,2024-08-01,2024-07-31,0,0,0,0,1,1
2,id01,2024-08-02,2024-08-01,0,0,1,0,1,1
3,id01,2024-08-03,2024-08-02,1,0,0,1,1,1
4,id01,2024-08-04,2024-08-03,1,1,1,1,1,1


In [ ]:
train_file_name3 = "submission_0.6383372929850419_catboost.csv"
submission_df3 = pd.read_csv(path+train_file_name3, encoding='utf8')

In [ ]:
submission_df3.rename(columns = {'Q1' : 'Target3_Q1'}, inplace = True)
submission_df3.rename(columns = {'Q2' : 'Target3_Q2'}, inplace = True)
submission_df3.rename(columns = {'Q3' : 'Target3_Q3'}, inplace = True)
submission_df3.rename(columns = {'S1' : 'Target3_S1'}, inplace = True)
submission_df3.rename(columns = {'S2' : 'Target3_S2'}, inplace = True)
submission_df3.rename(columns = {'S3' : 'Target3_S3'}, inplace = True)

In [ ]:
submission_df3.head()

,subject_id,sleep_date,lifelog_date,Target3_Q1,Target3_Q2,Target3_Q3,Target3_S1,Target3_S2,Target3_S3
0,id01,2024-07-31,2024-07-30,0,1,1,1,1,1
1,id01,2024-08-01,2024-07-31,0,0,0,0,1,1
2,id01,2024-08-02,2024-08-01,0,0,1,0,1,1
3,id01,2024-08-03,2024-08-02,1,0,0,1,1,1
4,id01,2024-08-04,2024-08-03,1,1,1,1,1,1


In [ ]:
df_merge_how_left = pd.merge(submission_df, submission_df2, how='left',on=['subject_id', 'sleep_date', 'lifelog_date'])

In [ ]:
df_merge_how_left.head()

,subject_id,sleep_date,lifelog_date,Target1_Q1,Target1_Q2,Target1_Q3,Target1_S1,Target1_S2,Target1_S3,Target2_Q1,Target2_Q2,Target2_Q3,Target2_S1,Target2_S2,Target2_S3
0,id01,2024-07-31,2024-07-30,0,1,0,1,1,1,0,1,0,1,1,1
1,id01,2024-08-01,2024-07-31,0,1,0,0,1,1,0,0,0,0,1,1
2,id01,2024-08-02,2024-08-01,0,0,1,0,0,1,0,0,1,0,1,1
3,id01,2024-08-03,2024-08-02,1,0,0,1,1,1,1,0,0,1,1,1
4,id01,2024-08-04,2024-08-03,1,1,1,1,1,1,1,1,1,1,1,1


In [ ]:
submission_df = pd.merge(df_merge_how_left, submission_df3, how='left',on=['subject_id', 'sleep_date', 'lifelog_date'])

In [ ]:
submission_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250 entries, 0 to 249
Data columns (total 21 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   subject_id    250 non-null    object
 1   sleep_date    250 non-null    object
 2   lifelog_date  250 non-null    object
 3   Target1_Q1    250 non-null    int64 
 4   Target1_Q2    250 non-null    int64 
 5   Target1_Q3    250 non-null    int64 
 6   Target1_S1    250 non-null    int64 
 7   Target1_S2    250 non-null    int64 
 8   Target1_S3    250 non-null    int64 
 9   Target2_Q1    250 non-null    int64 
 10  Target2_Q2    250 non-null    int64 
 11  Target2_Q3    250 non-null    int64 
 12  Target2_S1    250 non-null    int64 
 13  Target2_S2    250 non-null    int64 
 14  Target2_S3    250 non-null    int64 
 15  Target3_Q1    250 non-null    int64 
 16  Target3_Q2    250 non-null    int64 
 17  Target3_Q3    250 non-null    int64 
 18  Target3_S1    250 non-null    int64 
 19  Target3_

In [ ]:
print(submission_df['Target1_Q1'].value_counts())
print(submission_df['Target1_Q2'].value_counts())
print(submission_df['Target1_Q3'].value_counts())
print(submission_df['Target1_S1'].value_counts())
print(submission_df['Target1_S2'].value_counts())
print(submission_df['Target1_S3'].value_counts())

Target1_Q1
1    132
0    118
Name: count, dtype: int64
Target1_Q2
1    154
0     96
Name: count, dtype: int64
Target1_Q3
1    168
0     82
Name: count, dtype: int64
Target1_S1
1    147
0     71
2     32
Name: count, dtype: int64
Target1_S2
1    168
0     82
Name: count, dtype: int64
Target1_S3
1    166
0     84
Name: count, dtype: int64


In [ ]:
print(submission_df['Target2_Q1'].value_counts())
print(submission_df['Target2_Q2'].value_counts())
print(submission_df['Target2_Q3'].value_counts())
print(submission_df['Target2_S1'].value_counts())
print(submission_df['Target2_S2'].value_counts())
print(submission_df['Target2_S3'].value_counts())

Target2_Q1
1    131
0    119
Name: count, dtype: int64
Target2_Q2
1    154
0     96
Name: count, dtype: int64
Target2_Q3
1    173
0     77
Name: count, dtype: int64
Target2_S1
1    147
0     76
2     27
Name: count, dtype: int64
Target2_S2
1    167
0     83
Name: count, dtype: int64
Target2_S3
1    174
0     76
Name: count, dtype: int64


In [ ]:
print(submission_df['Target3_Q1'].value_counts())
print(submission_df['Target3_Q2'].value_counts())
print(submission_df['Target3_Q3'].value_counts())
print(submission_df['Target3_S1'].value_counts())
print(submission_df['Target3_S2'].value_counts())
print(submission_df['Target3_S3'].value_counts())

Target3_Q1
1    134
0    116
Name: count, dtype: int64
Target3_Q2
1    157
0     93
Name: count, dtype: int64
Target3_Q3
1    175
0     75
Name: count, dtype: int64
Target3_S1
1    159
0     65
2     26
Name: count, dtype: int64
Target3_S2
1    169
0     81
Name: count, dtype: int64
Target3_S3
1    163
0     87
Name: count, dtype: int64


In [ ]:
from collections import Counter
for idx, row in submission_df.iterrows():
  Target1_Q1 = row["Target1_Q1"]
  Target2_Q1 = row["Target2_Q1"]
  Target3_Q1 = row["Target3_Q1"]

  z = [Target1_Q1, Target2_Q1,  Target3_Q1]
  dic_list = Counter(z)
  max_value = max(dic_list, key=dic_list.get)
  submission_df.loc[idx, 'Q1'] = max_value

In [ ]:
print(submission_df['Q1'].value_counts())
print(submission_df['Q1'].count())

Q1
1.0    131
0.0    119
Name: count, dtype: int64
250


In [ ]:
for idx, row in submission_df.iterrows():
  Target1_Q2 = row["Target1_Q2"]
  Target2_Q2 = row["Target2_Q2"]
  Target3_Q2 = row["Target3_Q2"]

  z = [Target1_Q2, Target2_Q2,  Target3_Q2]
  dic_list = Counter(z)
  max_value = max(dic_list, key=dic_list.get)
  submission_df.loc[idx, 'Q2'] = max_value

In [ ]:
print(submission_df['Q2'].value_counts())
print(submission_df['Q2'].count())

Q2
1.0    157
0.0     93
Name: count, dtype: int64
250


In [ ]:
for idx, row in submission_df.iterrows():
  Target1_Q3 = row["Target1_Q3"]
  Target2_Q3 = row["Target2_Q3"]
  Target3_Q3 = row["Target3_Q3"]

  z = [Target1_Q3, Target2_Q3,  Target3_Q3]
  dic_list = Counter(z)
  max_value = max(dic_list, key=dic_list.get)
  submission_df.loc[idx, 'Q3'] = max_value

In [ ]:
print(submission_df['Q3'].value_counts())
print(submission_df['Q3'].count())

Q3
1.0    170
0.0     80
Name: count, dtype: int64
250


In [ ]:
submission_df.head(100)

,subject_id,sleep_date,lifelog_date,Target1_Q1,Target1_Q2,Target1_Q3,Target1_S1,Target1_S2,Target1_S3,Target2_Q1,...,Target2_S1,Target2_S2,Target2_S3,Target3_Q1,Target3_Q2,Target3_Q3,Target3_S1,Target3_S2,Target3_S3,Q1
0,id01,2024-07-31,2024-07-30,0,1,0,1,1,1,0,...,1,1,1,0,1,1,1,1,1,0.0
1,id01,2024-08-01,2024-07-31,0,1,0,0,1,1,0,...,0,1,1,0,0,0,0,1,1,0.0
2,id01,2024-08-02,2024-08-01,0,0,1,0,0,1,0,...,0,1,1,0,0,1,0,1,1,0.0
3,id01,2024-08-03,2024-08-02,1,0,0,1,1,1,1,...,1,1,1,1,0,0,1,1,1,1.0
4,id01,2024-08-04,2024-08-03,1,1,1,1,1,1,1,...,1,1,1,1,1,1,1,1,1,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
95,id04,2024-10-16,2024-10-15,1,1,1,2,1,0,1,...,2,1,0,1,1,1,2,1,0,1.0
96,id04,2024-10-17,2024-10-16,1,1,1,1,0,0,0,...,1,1,0,1,1,1,1,0,0,1.0
97,id04,2024-10-18,2024-10-17,1,1,1,1,1,0,1,...,2,1,0,1,1,1,1,1,0,1.0
98,id04,2024-10-19,2024-10-18,1,1,1,1,1,0,1,...,1,0,0,1,1,1,1,1,0,1.0


In [ ]:
for idx, row in submission_df.iterrows():
  Target1_S1 = row["Target1_S1"]
  Target2_S1 = row["Target2_S1"]
  Target3_S1 = row["Target3_S1"]

  z = [Target1_S1, Target2_S1,  Target3_S1]
  dic_list = Counter(z)
  max_value = max(dic_list, key=dic_list.get)
  submission_df.loc[idx, 'S1'] = max_value

In [ ]:
print(submission_df['S1'].value_counts())
print(submission_df['S1'].count())

S1
1.0    151
0.0     71
2.0     28
Name: count, dtype: int64
250


In [ ]:
for idx, row in submission_df.iterrows():
  Target1_S2 = row["Target1_S2"]
  Target2_S2 = row["Target2_S2"]
  Target3_S2 = row["Target3_S2"]

  z = [Target1_S2, Target2_S2,  Target3_S2]
  dic_list = Counter(z)
  max_value = max(dic_list, key=dic_list.get)
  submission_df.loc[idx, 'S2'] = max_value

In [ ]:
print(submission_df['S2'].value_counts())
print(submission_df['S2'].count())

S2
1.0    169
0.0     81
Name: count, dtype: int64
250


In [ ]:
for idx, row in submission_df.iterrows():
  Target1_S3 = row["Target1_S3"]
  Target2_S3 = row["Target2_S3"]
  Target3_S3 = row["Target3_S3"]

  z = [Target1_S3, Target2_S3,  Target3_S3]
  dic_list = Counter(z)
  max_value = max(dic_list, key=dic_list.get)
  submission_df.loc[idx, 'S3'] = max_value

In [ ]:
print(submission_df['S3'].value_counts())
print(submission_df['S3'].count())

S3
1.0    167
0.0     83
Name: count, dtype: int64
250


In [ ]:
submission_df['Q1'] = submission_df['Q1'].astype(int)
submission_df['Q2'] = submission_df['Q2'].astype(int)
submission_df['Q3'] = submission_df['Q3'].astype(int)
submission_df['S1'] = submission_df['S1'].astype(int)
submission_df['S2'] = submission_df['S2'].astype(int)
submission_df['S3'] = submission_df['S3'].astype(int)

In [ ]:
submission_df.describe()

,Target1_Q1,Target1_Q2,Target1_Q3,Target1_S1,Target1_S2,Target1_S3,Target2_Q1,Target2_Q2,Target2_Q3,Target2_S1,...,Target3_Q3,Target3_S1,Target3_S2,Target3_S3,Q1,Q2,Q3,S1,S2,S3
count,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,...,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000,250.000000
mean,0.528000,0.616000,0.672000,0.844000,0.672000,0.664000,0.524000,0.616000,0.692000,0.804000,...,0.700000,0.844000,0.676000,0.652000,0.524000,0.628000,0.680000,0.828000,0.676000,0.668000
std,0.500217,0.487334,0.470427,0.623876,0.470427,0.473286,0.500426,0.487334,0.462593,0.612441,...,0.459177,0.583976,0.468939,0.477292,0.500426,0.484308,0.467412,0.606537,0.468939,0.471876
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,...,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000
75%,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,...,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000
max,1.000000,1.000000,1.000000,2.000000,1.000000,1.000000,1.000000,1.000000,1.000000,2.000000,...,1.000000,2.000000,1.000000,1.000000,1.000000,1.000000,1.000000,2.000000,1.000000,1.000000


In [ ]:
submission_file_name = "submission_3_hard_voting.csv"

submission_df2 = pd.DataFrame()
submission_df2["subject_id"] = submission_df['subject_id']
submission_df2["sleep_date"] = submission_df['sleep_date']
submission_df2["lifelog_date"] = submission_df['lifelog_date']
submission_df2["Q1"] = submission_df['Q1']
submission_df2["Q2"] = submission_df['Q2']
submission_df2["Q3"] = submission_df['Q3']
submission_df2["S1"] = submission_df['S1']
submission_df2["S2"] = submission_df['S2']
submission_df2["S3"] = submission_df['S3']
submission_df2.to_csv(path+submission_file_name, index=False)
print(f"Saved 'finished")

Saved 'finished


In [ ]:
submission_df2.head()

,subject_id,sleep_date,lifelog_date,Q1,Q2,Q3,S1,S2,S3
0,id01,2024-07-31,2024-07-30,0,1,0,1,1,1
1,id01,2024-08-01,2024-07-31,0,0,0,0,1,1
2,id01,2024-08-02,2024-08-01,0,0,1,0,1,1
3,id01,2024-08-03,2024-08-02,1,0,0,1,1,1
4,id01,2024-08-04,2024-08-03,1,1,1,1,1,1


2.0    392
0.0    344
5.0    339
4.0    306
3.0    282
1.0    218 -> 0.52
->순위 중복이더라도 타겟2 먼저 선택함


2.0    415
0.0    335
5.0    331
4.0    330
3.0    270
1.0    200 -> 타켓 1 먼저 선택함